src/ExecutionLoop.php

```diff
@@ -29,8 +29,10 @@ public function run(Shell $shell)
     {
         $this->loadIncludes($shell);
 
-        $closure = new ExecutionLoopClosure($shell);
-        $closure->execute();
+        if ($shell->isInteractive()) {
+            $closure = new ExecutionLoopClosure($shell);
+            $closure->execute();
```

Comment:
Yeah, I see your point. I'll try something.

Reply:
The ExecutionLoop is really not looping.. it's just setting things up for the ExecutionLoopClosure, which is where the loop actually is. I'm wondering if ExecutionClosure could be used as the non-interactive closure, and ExecutionLoopClosure for interactive.

So shell could do:
```php
  $executionClosure = $this->isInteractive ? new ExecutionLoopClosure($this) : new ExecutionClosure($this);
  $this->loop->run($this, $executionClosure);
```

`$this->loop` would prob get renamed... `executor` or something.

Reply:
in the case of non-interactive execution, you don't want ExecutionClosure either. you just want the `loadIncludes` behavior from ExecutionLoop. so maybe the thing to change is all the way up in `Shell::doRun`, and make it `loadIncludes` rather than the entire `beforeRun`/`run`/`afterRun` business?